great_expectations/expectations/metrics/util.py

```diff
@@ -608,26 +615,44 @@ def column_reflection_fallback(
             # noinspection PyUnresolvedReferences
             if dialect.name.lower() == GXSqlDialect.REDSHIFT:
                 # Redshift needs temp tables to be declared as text
-                query: Select = (
+                query = (
                     sa.select([sa.text("*")]).select_from(sa.text(selectable)).limit(1)
                 )
             else:
-                query: Select = (
-                    sa.select([sa.text("*")]).select_from(selectable).limit(1)
-                )
+                query = sa.select([sa.text("*")]).select_from(selectable).limit(1)
         result_object = sqlalchemy_engine.execute(query)
         # noinspection PyProtectedMember
         col_names: List[str] = result_object._metadata.keys
         col_info_dict_list = [{"name": col_name} for col_name in col_names]
     return col_info_dict_list
 
 
+@overload
 def get_dbms_compatible_column_names(
-    column_names: Union[List[str], str],
-    batch_columns_list: List[Union[str, quoted_name]],
+    column_names: str,
+    batch_columns_list: List[str],
+    execution_engine: ExecutionEngine,
+    error_message_template: str = ...,
+) -> str:
+    ...
+
+
+@overload
+def get_dbms_compatible_column_names(
+    column_names: List[str],
+    batch_columns_list: List[str],
+    execution_engine: ExecutionEngine,
+    error_message_template: str = ...,
+) -> List[str]:
+    ...
+
+
+def get_dbms_compatible_column_names(
+    column_names: List[str] | str,
+    batch_columns_list: List[str],
     execution_engine: ExecutionEngine,
     error_message_template: str = 'Error: The column "{column_name:s}" in BatchData does not exist.',
-) -> Union[List[Union[str, quoted_name]], Union[str, quoted_name]]:
```

Comment:
Any idea what this `quoted_name` annotation was meant to represent?

Reply:
`sqlalchemy.sql.quoted_name`

Reply:
Thanks, will update after lunch

Reply:
`quoted_name` is an `SQLAlchemy` type (which is critical to us).